### PR TITLE
Dynamic countries page and component renaming

### PR DESCRIPTION
**Pull request by aedanmc**
Changes made in this PR:
1. Update the static version of the countries page to have dynamic data fetching 
2. Introduce a limited filter tool to see a country of interest
3. Rename front-end components to have better distinction between pages and components
4. Integrate into the existing system so that navigation between pages is possible
5. Data fetching displays meaningful info about countries after searching

General comments: I'm requesting everyone's feedback since I feel like the styling could be a little prettier for the detailed country data, but I wanted to get a draft up since I've been working on this all day and want to take a break.

### CODE REVIEW

**Review by C-Christensen — APPROVED** [review]
LGTM

File: front-end/src/components/countries-search-page/CountrySearchPage.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/* eslint-disable max-len */
+import * as React from 'react';
+// import Stack from '@mui/material/Stack';
```
Comment (mik3ycodez) [inline front-end/src/components/countries-search-page/CountrySearchPage.js:3]: Lots of commented out code -- should this be removed? 
- import statements
- lines 45-68

**Review by mik3ycodez — CHANGES_REQUESTED** [review]

**Review by mik3ycodez — APPROVED** [review]
LGTM

**Review by matthewdbroom13 — APPROVED** [review]

**Comment by matthewdbroom13** [comment]
LGTM!

**Review by mzuniga-1833381 — APPROVED** [review]
LGTM!

**Pull request merged**